src/DeckViewer.jsx

```diff
@@ -34,11 +39,16 @@ export default function DeckViewer() {
       </div>
 
       <div style={controlsStyle}>
-        <input placeholder="Filter (e.g., A, hearts, 10♣)" value={query} onChange={e => setQuery(e.target.value)} style={inputStyle} />
+        <input
+          placeholder="Filter (e.g., A, hearts, 10♣)"
+          value={query}
+          onChange={e => setQuery(e.target.value)}
+          style={inputStyle}
+        />
         <button style={btnStyle} onClick={sortByRank}>
           Sort by Rank
         </button>
-        <button style={btnStyle} onClick={shuffleDeck}>
+        <button style={btnStyle} onClick={shuffleStoreDeck()}>
```

Comment:
Remove the parentheses so that the function isn't invoked on render. It was correct the way it was before.